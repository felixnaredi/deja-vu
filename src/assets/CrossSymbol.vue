<template>
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 109.37 109.37">
    <g>
      <g>
        <circle class="cls-1" cx="54.69" cy="54.69" r="54.69" />
        <g>
          <line class="cls-2" x1="32.52" y1="32.52" x2="76.85" y2="76.85" />
          <line class="cls-2" x1="76.85" y1="32.52" x2="32.52" y2="76.85" />
        </g>
      </g>
    </g>
  </svg>
</template>

<script>
export default {};
</script>

<style scoped>
.cls-1 {
  fill: rgb(239 68 68);
}

.cls-2 {
  fill: none;
  stroke: rgb(254 242 242);
  stroke-miterlimit: 10;
  stroke-width: 16px;
}
</style>
