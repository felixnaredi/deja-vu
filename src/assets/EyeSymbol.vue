<template>
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 120 100">
    <g>
      <ellipse
        class="fill-transparent stroke-black stroke-[6px]"
        cx="54"
        cy="50"
        rx="50"
        ry="25"
      />
      <g>
        <circle
          class="fill-slate-500 stroke-black stroke-[6px]"
          cx="53"
          cy="43"
          r="18"
        />
        <circle class="fill-white" cx="60" cy="38" r="8" />
      </g>
    </g>
  </svg>
</template>

<script>
export default {};
</script>
