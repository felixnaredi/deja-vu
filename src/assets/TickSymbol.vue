<template>
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 109.38 109.38">
    <g>
      <g>
        <circle class="cls-1" cx="54.69" cy="54.69" r="54.69" />
        <polyline class="cls-2" points="22.32 52.84 49.06 76.32 87.06 33.05" />
      </g>
    </g>
  </svg>
</template>

<script>
export default {};
</script>

<style scoped>
.cls-1 {
  fill: rgb(34 197 94);
}

.cls-2 {
  fill: none;
  stroke: #fff;
  stroke-miterlimit: 10;
  stroke-width: 16px;
}
</style>
