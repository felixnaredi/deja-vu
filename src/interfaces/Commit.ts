import SeenUnseen from "@/interfaces/SeenUnseen";

interface Commit {
  element: () => string;
  actual: () => SeenUnseen;
  guess: () => SeenUnseen;
  correct: () => boolean;
}

export default Commit;
