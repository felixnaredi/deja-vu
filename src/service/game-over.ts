import { GameOver, Commit } from "deja-vu-wasm";

function GameOverIterator(gameOver: GameOver): Iterable<Commit> {
  const iterator = gameOver.iterator();
  return {
    [Symbol.iterator]: function* () {
      do {
        const x = iterator.next();
        if (x == null) {
          break;
        } else {
          yield x;
        }
      } while (true);
    },
  };
}

export function commits(gameOver: GameOver): Commit[] {
  return Array.from(GameOverIterator(gameOver));
}
