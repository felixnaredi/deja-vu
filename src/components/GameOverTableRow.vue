<template>
  <div
    class="grid grid-cols-8 gap-4 border-b-2"
    :class="{ 'bg-red-50': !commit.correct() }"
  >
    <div v-if="commit.actual().isSeen()" class="h-full justify-self-center">
      <eye-symbol class="h-full w-5" />
    </div>
    <div v-else></div>
    <div class="col-span-5 ml-4">{{ commit.element() }}</div>
    <div class="justify-self-center">
      <div v-if="commit.guess().isSeen()" class="h-full">
        <tick-symbol v-if="commit.correct()" class="w-5 h-full" />
        <cross-symbol v-else class="w-5 h-full" />
      </div>
      <div v-else></div>
    </div>
    <div v-if="commit.guess().isUnseen()" class="h-full justify-self-center">
      <tick-symbol v-if="commit.correct()" class="w-5 h-full" />
      <cross-symbol v-else class="w-5 h-full" />
    </div>
    <div v-else></div>
  </div>
</template>

<script>
import TickSymbol from "@/assets/TickSymbol.vue";
import CrossSymbol from "@/assets/CrossSymbol.vue";
import EyeSymbol from "@/assets/EyeSymbol.vue";
import Commit from "@/interfaces/_Commit";

export default {
  components: { TickSymbol, CrossSymbol, EyeSymbol },
  props: {
    commit: Commit,
  },
};
</script>
