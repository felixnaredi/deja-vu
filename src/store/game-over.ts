import { defineStore } from "pinia";
import { commits } from "@/service/game-over";
import { GameOver, Commit } from "deja-vu-wasm";

interface State {
  gameOver: undefined | GameOver;
}

export const useGameOverStore = defineStore("game-over", {
  state: () =>
    ({
      gameOver: undefined,
    } as State),
  getters: {
    commits(): undefined | Commit[] {
      if (this.gameOver == undefined) {
        return undefined;
      } else {
        return commits(this.gameOver);
      }
    },
    score(): undefined | number {
      return this.gameOver?.score();
    },
    lives(): undefined | number {
      return this.gameOver?.lives();
    },
  },
  actions: {
    setGameOver(gameOver: GameOver) {
      this.gameOver = gameOver;
    },
  },
});
