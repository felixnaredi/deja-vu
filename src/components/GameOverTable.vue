<template>
  <div>
    <div class="grid grid-cols-8 gap-4 mb-3">
      <div class="col-span-6"></div>
      <span class="justify-self-center underline">déjà vu</span>
      <span class="justify-self-center underline">nouveau</span>
    </div>
    <game-over-table-row
      v-for="commit in commits"
      :key="commit.index"
      :commit="commit.value"
    />
  </div>
</template>

<script>
import { useGameOverStore } from "@/store/game-over";
import GameOverTableRow from "@/components/GameOverTableRow.vue";

export default {
  components: { GameOverTableRow },
  computed: {
    commits: () => {
      return useGameOverStore().commits?.map((value, index) => ({
        index,
        value,
      }));
    },
  },
};
</script>
