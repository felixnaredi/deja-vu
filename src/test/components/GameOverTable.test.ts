import GameOverTable from "@/components/GameOverTable.vue";
import GameOverTableRow from "@/components/GameOverTableRow.vue";
import { mount, shallowMount, VueWrapper } from "@vue/test-utils";
import { createTestingPinia } from "@pinia/testing";
import { useGameOverStore } from "@/store/game-over";
import Commit from "@/interfaces/Commit";
import { generateCommit, Seen, Unseen } from "@/test/commit-mock";
import CrossSymbol from "@/assets/CrossSymbol.vue";
import TickSymbol from "@/assets/TickSymbol.vue";
import EyeSymbol from "@/assets/EyeSymbol.vue";
import { setActivePinia } from "pinia";

function createWrapper(
  mountFunction: typeof shallowMount | typeof mount,
  commits: undefined | Array<Commit>
): VueWrapper<any> {
  const pinia = createTestingPinia();
  setActivePinia(pinia);
  const gameOverStore = useGameOverStore(pinia);

  // `createTestingPinia` makes `commits` into a writable property
  // @ts-ignore
  gameOverStore.commits = commits;

  return mountFunction(GameOverTable);
}

function shallowGameOverTable(
  commits: undefined | Array<Commit>
): VueWrapper<any> {
  return createWrapper(shallowMount, commits);
}

function mountGameOverTable(
  commits: undefined | Array<Commit>
): VueWrapper<any> {
  return createWrapper(mount, commits);
}

describe("unit tests", () => {
  test("no rows, only header, when commit list is undefined", () => {
    // TODO:
    //   Muting this output. Could be of interest to solve why the warning in triggered and prevent
    //   it.
    // ```
    // stderr | src/test/components/GameOverTable.test.ts > unit tests > no rows, only header, when commit list is undefined
    // [Vue warn]: injection "Symbol(pinia)" not found.
    //   at <GameOverTable ref="VTU_COMPONENT" >
    //   at <VTUROOT>
    // ```
    vi.spyOn(console, "warn").mockImplementation(() => {});

    const wrapper = shallowGameOverTable(undefined);
    expect(wrapper.html()).toContain("déjà vu");
    expect(wrapper.html()).toContain("nouveau");
    expect(wrapper.findComponent(GameOverTableRow).exists()).not.toBeTruthy();
  });

  test("no rows, only header, when commit list is empty", () => {
    // TODO:
    //   Muting this output. Could be of interest to solve why the warning in triggered and prevent
    //   it.
    // ```
    // stderr | src/test/components/GameOverTable.test.ts > unit tests > no rows, only header, when commit list is empty
    // stderr | src/test/components/GameOverTable.test.ts > unit tests > no rows, only header, when commit list is empty
    // [Vue warn]: injection "Symbol(pinia)" not found.
    //   at <GameOverTable ref="VTU_COMPONENT" >
    //   at <VTUROOT>
    //
    // [Vue warn]: injection "Symbol(pinia)" not found.
    //   at <GameOverTable ref="VTU_COMPONENT" >
    //   at <VTUROOT>
    // ```
    vi.spyOn(console, "warn").mockImplementation(() => {});

    const wrapper = shallowGameOverTable([]);
    expect(wrapper.html()).toContain("déjà vu");
    expect(wrapper.html()).toContain("nouveau");
    expect(wrapper.findComponent(GameOverTableRow).exists()).not.toBeTruthy();
  });

  test("no rows, only header, when commit list is empty", () => {
    const wrapper = mountGameOverTable([
      generateCommit({
        seed: 1818439690026315,
        actual: new Seen(),
        guess: new Seen(),
      }),
      generateCommit({
        seed: 881250011915912,
        actual: new Unseen(),
        guess: new Seen(),
      }),
      generateCommit({
        seed: 228821152228297,
        actual: new Seen(),
        guess: new Unseen(),
      }),
      generateCommit({
        seed: 8034100225224008,
        actual: new Unseen(),
        guess: new Unseen(),
      }),
    ]);
    expect(wrapper.html()).toContain("déjà vu");
    expect(wrapper.html()).toContain("nouveau");
    expect(wrapper.findAllComponents(GameOverTableRow).length).toBe(4);
    expect(wrapper.findAllComponents(CrossSymbol).length).toBe(2);
    expect(wrapper.findAllComponents(TickSymbol).length).toBe(2);
    expect(wrapper.findAllComponents(EyeSymbol).length).toBe(2);
  });
});

describe("snapshots", () => {
  test("snapshot of `GameOverTable` with `commits` set to `undefined`", () => {
    const wrapper = shallowGameOverTable(undefined);
    expect(wrapper.html()).toMatchSnapshot();
  });
  test("snapshot of `GameOverTable` with `commits` set to empty list", () => {
    const wrapper = shallowGameOverTable([]);
    expect(wrapper.html()).toMatchSnapshot();
  });
  test("snapshot of `GameOverTable` with `commits` some generated `Commit`s", () => {
    const wrapper = mountGameOverTable(
      [
        1835952554073773, 76584337047733, 158407870645769, 3806636883036028,
        999837377973796, 1181176957149847, 147709579785058, 1645148002455520,
        1571701253944690, 566009344100606, 370914558206118,
      ].map((seed) => generateCommit({ seed }))
    );
    expect(wrapper.html()).toMatchSnapshot();
  });
});
