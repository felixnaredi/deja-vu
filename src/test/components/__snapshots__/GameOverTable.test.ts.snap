// Vitest Snapshot v1

exports[`snapshots > snapshot of \`GameOverTable\` with \`commits\` set to \`undefined\` 1`] = `
"<div>
  <div class=\\"grid grid-cols-8 gap-4 mb-3\\">
    <div class=\\"col-span-6\\"></div><span class=\\"justify-self-center underline\\">déjà vu</span><span class=\\"justify-self-center underline\\">nouveau</span>
  </div>
</div>"
`;

exports[`snapshots > snapshot of \`GameOverTable\` with \`commits\` set to empty list 1`] = `
"<div>
  <div class=\\"grid grid-cols-8 gap-4 mb-3\\">
    <div class=\\"col-span-6\\"></div><span class=\\"justify-self-center underline\\">déjà vu</span><span class=\\"justify-self-center underline\\">nouveau</span>
  </div>
</div>"
`;

exports[`snapshots > snapshot of \`GameOverTable\` with \`commits\` some generated \`Commit\`s 1`] = `
"<div>
  <div class=\\"grid grid-cols-8 gap-4 mb-3\\">
    <div class=\\"col-span-6\\"></div><span class=\\"justify-self-center underline\\">déjà vu</span><span class=\\"justify-self-center underline\\">nouveau</span>
  </div>
  <div class=\\"grid grid-cols-8 gap-4 border-b-2\\">
    <div class=\\"h-full justify-self-center\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 120 100\\" class=\\"h-full w-5\\">
        <g>
          <ellipse class=\\"fill-transparent stroke-black stroke-[6px]\\" cx=\\"54\\" cy=\\"50\\" rx=\\"50\\" ry=\\"25\\"></ellipse>
          <g>
            <circle class=\\"fill-slate-500 stroke-black stroke-[6px]\\" cx=\\"53\\" cy=\\"43\\" r=\\"18\\"></circle>
            <circle class=\\"fill-white\\" cx=\\"60\\" cy=\\"38\\" r=\\"8\\"></circle>
          </g>
        </g>
      </svg></div>
    <div class=\\"col-span-5 ml-4\\">MiulY0rX1QNlqWMcMj2CJ4g</div>
    <div class=\\"justify-self-center\\">
      <div class=\\"h-full\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 109.38 109.38\\" class=\\"w-5 h-full\\" data-v-34d8da4e=\\"\\">
          <g data-v-34d8da4e=\\"\\">
            <g data-v-34d8da4e=\\"\\">
              <circle class=\\"cls-1\\" cx=\\"54.69\\" cy=\\"54.69\\" r=\\"54.69\\" data-v-34d8da4e=\\"\\"></circle>
              <polyline class=\\"cls-2\\" points=\\"22.32 52.84 49.06 76.32 87.06 33.05\\" data-v-34d8da4e=\\"\\"></polyline>
            </g>
          </g>
        </svg></div>
    </div>
    <div></div>
  </div>
  <div class=\\"grid grid-cols-8 gap-4 border-b-2\\">
    <div></div>
    <div class=\\"col-span-5 ml-4\\">NAzo0QcF6C8QxnMRMgU3</div>
    <div class=\\"justify-self-center\\">
      <div></div>
    </div>
    <div class=\\"h-full justify-self-center\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 109.38 109.38\\" class=\\"w-5 h-full\\" data-v-34d8da4e=\\"\\">
        <g data-v-34d8da4e=\\"\\">
          <g data-v-34d8da4e=\\"\\">
            <circle class=\\"cls-1\\" cx=\\"54.69\\" cy=\\"54.69\\" r=\\"54.69\\" data-v-34d8da4e=\\"\\"></circle>
            <polyline class=\\"cls-2\\" points=\\"22.32 52.84 49.06 76.32 87.06 33.05\\" data-v-34d8da4e=\\"\\"></polyline>
          </g>
        </g>
      </svg></div>
  </div>
  <div class=\\"grid grid-cols-8 gap-4 border-b-2\\">
    <div></div>
    <div class=\\"col-span-5 ml-4\\">UQ7c9HRadt6juO9</div>
    <div class=\\"justify-self-center\\">
      <div></div>
    </div>
    <div class=\\"h-full justify-self-center\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 109.38 109.38\\" class=\\"w-5 h-full\\" data-v-34d8da4e=\\"\\">
        <g data-v-34d8da4e=\\"\\">
          <g data-v-34d8da4e=\\"\\">
            <circle class=\\"cls-1\\" cx=\\"54.69\\" cy=\\"54.69\\" r=\\"54.69\\" data-v-34d8da4e=\\"\\"></circle>
            <polyline class=\\"cls-2\\" points=\\"22.32 52.84 49.06 76.32 87.06 33.05\\" data-v-34d8da4e=\\"\\"></polyline>
          </g>
        </g>
      </svg></div>
  </div>
  <div class=\\"grid grid-cols-8 gap-4 border-b-2\\">
    <div class=\\"h-full justify-self-center\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 120 100\\" class=\\"h-full w-5\\">
        <g>
          <ellipse class=\\"fill-transparent stroke-black stroke-[6px]\\" cx=\\"54\\" cy=\\"50\\" rx=\\"50\\" ry=\\"25\\"></ellipse>
          <g>
            <circle class=\\"fill-slate-500 stroke-black stroke-[6px]\\" cx=\\"53\\" cy=\\"43\\" r=\\"18\\"></circle>
            <circle class=\\"fill-white\\" cx=\\"60\\" cy=\\"38\\" r=\\"8\\"></circle>
          </g>
        </g>
      </svg></div>
    <div class=\\"col-span-5 ml-4\\">PMLsvfjgEHvT</div>
    <div class=\\"justify-self-center\\">
      <div class=\\"h-full\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 109.38 109.38\\" class=\\"w-5 h-full\\" data-v-34d8da4e=\\"\\">
          <g data-v-34d8da4e=\\"\\">
            <g data-v-34d8da4e=\\"\\">
              <circle class=\\"cls-1\\" cx=\\"54.69\\" cy=\\"54.69\\" r=\\"54.69\\" data-v-34d8da4e=\\"\\"></circle>
              <polyline class=\\"cls-2\\" points=\\"22.32 52.84 49.06 76.32 87.06 33.05\\" data-v-34d8da4e=\\"\\"></polyline>
            </g>
          </g>
        </svg></div>
    </div>
    <div></div>
  </div>
  <div class=\\"grid grid-cols-8 gap-4 border-b-2 bg-red-50\\">
    <div class=\\"h-full justify-self-center\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 120 100\\" class=\\"h-full w-5\\">
        <g>
          <ellipse class=\\"fill-transparent stroke-black stroke-[6px]\\" cx=\\"54\\" cy=\\"50\\" rx=\\"50\\" ry=\\"25\\"></ellipse>
          <g>
            <circle class=\\"fill-slate-500 stroke-black stroke-[6px]\\" cx=\\"53\\" cy=\\"43\\" r=\\"18\\"></circle>
            <circle class=\\"fill-white\\" cx=\\"60\\" cy=\\"38\\" r=\\"8\\"></circle>
          </g>
        </g>
      </svg></div>
    <div class=\\"col-span-5 ml-4\\">iiXWRB8vfV6vIbgTMV9W</div>
    <div class=\\"justify-self-center\\">
      <div></div>
    </div>
    <div class=\\"h-full justify-self-center\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 109.37 109.37\\" class=\\"w-5 h-full\\" data-v-57536c88=\\"\\">
        <g data-v-57536c88=\\"\\">
          <g data-v-57536c88=\\"\\">
            <circle class=\\"cls-1\\" cx=\\"54.69\\" cy=\\"54.69\\" r=\\"54.69\\" data-v-57536c88=\\"\\"></circle>
            <g data-v-57536c88=\\"\\">
              <line class=\\"cls-2\\" x1=\\"32.52\\" y1=\\"32.52\\" x2=\\"76.85\\" y2=\\"76.85\\" data-v-57536c88=\\"\\"></line>
              <line class=\\"cls-2\\" x1=\\"76.85\\" y1=\\"32.52\\" x2=\\"32.52\\" y2=\\"76.85\\" data-v-57536c88=\\"\\"></line>
            </g>
          </g>
        </g>
      </svg></div>
  </div>
  <div class=\\"grid grid-cols-8 gap-4 border-b-2\\">
    <div></div>
    <div class=\\"col-span-5 ml-4\\">m1lyshsJGOk4dbmegCBf15n</div>
    <div class=\\"justify-self-center\\">
      <div></div>
    </div>
    <div class=\\"h-full justify-self-center\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 109.38 109.38\\" class=\\"w-5 h-full\\" data-v-34d8da4e=\\"\\">
        <g data-v-34d8da4e=\\"\\">
          <g data-v-34d8da4e=\\"\\">
            <circle class=\\"cls-1\\" cx=\\"54.69\\" cy=\\"54.69\\" r=\\"54.69\\" data-v-34d8da4e=\\"\\"></circle>
            <polyline class=\\"cls-2\\" points=\\"22.32 52.84 49.06 76.32 87.06 33.05\\" data-v-34d8da4e=\\"\\"></polyline>
          </g>
        </g>
      </svg></div>
  </div>
  <div class=\\"grid grid-cols-8 gap-4 border-b-2 bg-red-50\\">
    <div></div>
    <div class=\\"col-span-5 ml-4\\">Nm3mTaGlGhurDIuBKZKmj</div>
    <div class=\\"justify-self-center\\">
      <div class=\\"h-full\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 109.37 109.37\\" class=\\"w-5 h-full\\" data-v-57536c88=\\"\\">
          <g data-v-57536c88=\\"\\">
            <g data-v-57536c88=\\"\\">
              <circle class=\\"cls-1\\" cx=\\"54.69\\" cy=\\"54.69\\" r=\\"54.69\\" data-v-57536c88=\\"\\"></circle>
              <g data-v-57536c88=\\"\\">
                <line class=\\"cls-2\\" x1=\\"32.52\\" y1=\\"32.52\\" x2=\\"76.85\\" y2=\\"76.85\\" data-v-57536c88=\\"\\"></line>
                <line class=\\"cls-2\\" x1=\\"76.85\\" y1=\\"32.52\\" x2=\\"32.52\\" y2=\\"76.85\\" data-v-57536c88=\\"\\"></line>
              </g>
            </g>
          </g>
        </svg></div>
    </div>
    <div></div>
  </div>
  <div class=\\"grid grid-cols-8 gap-4 border-b-2 bg-red-50\\">
    <div class=\\"h-full justify-self-center\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 120 100\\" class=\\"h-full w-5\\">
        <g>
          <ellipse class=\\"fill-transparent stroke-black stroke-[6px]\\" cx=\\"54\\" cy=\\"50\\" rx=\\"50\\" ry=\\"25\\"></ellipse>
          <g>
            <circle class=\\"fill-slate-500 stroke-black stroke-[6px]\\" cx=\\"53\\" cy=\\"43\\" r=\\"18\\"></circle>
            <circle class=\\"fill-white\\" cx=\\"60\\" cy=\\"38\\" r=\\"8\\"></circle>
          </g>
        </g>
      </svg></div>
    <div class=\\"col-span-5 ml-4\\">6AjWuRQUpQ3KRDWL</div>
    <div class=\\"justify-self-center\\">
      <div></div>
    </div>
    <div class=\\"h-full justify-self-center\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 109.37 109.37\\" class=\\"w-5 h-full\\" data-v-57536c88=\\"\\">
        <g data-v-57536c88=\\"\\">
          <g data-v-57536c88=\\"\\">
            <circle class=\\"cls-1\\" cx=\\"54.69\\" cy=\\"54.69\\" r=\\"54.69\\" data-v-57536c88=\\"\\"></circle>
            <g data-v-57536c88=\\"\\">
              <line class=\\"cls-2\\" x1=\\"32.52\\" y1=\\"32.52\\" x2=\\"76.85\\" y2=\\"76.85\\" data-v-57536c88=\\"\\"></line>
              <line class=\\"cls-2\\" x1=\\"76.85\\" y1=\\"32.52\\" x2=\\"32.52\\" y2=\\"76.85\\" data-v-57536c88=\\"\\"></line>
            </g>
          </g>
        </g>
      </svg></div>
  </div>
  <div class=\\"grid grid-cols-8 gap-4 border-b-2 bg-red-50\\">
    <div class=\\"h-full justify-self-center\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 120 100\\" class=\\"h-full w-5\\">
        <g>
          <ellipse class=\\"fill-transparent stroke-black stroke-[6px]\\" cx=\\"54\\" cy=\\"50\\" rx=\\"50\\" ry=\\"25\\"></ellipse>
          <g>
            <circle class=\\"fill-slate-500 stroke-black stroke-[6px]\\" cx=\\"53\\" cy=\\"43\\" r=\\"18\\"></circle>
            <circle class=\\"fill-white\\" cx=\\"60\\" cy=\\"38\\" r=\\"8\\"></circle>
          </g>
        </g>
      </svg></div>
    <div class=\\"col-span-5 ml-4\\">M25zigo2PGIz8BiXYjnulAL</div>
    <div class=\\"justify-self-center\\">
      <div></div>
    </div>
    <div class=\\"h-full justify-self-center\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 109.37 109.37\\" class=\\"w-5 h-full\\" data-v-57536c88=\\"\\">
        <g data-v-57536c88=\\"\\">
          <g data-v-57536c88=\\"\\">
            <circle class=\\"cls-1\\" cx=\\"54.69\\" cy=\\"54.69\\" r=\\"54.69\\" data-v-57536c88=\\"\\"></circle>
            <g data-v-57536c88=\\"\\">
              <line class=\\"cls-2\\" x1=\\"32.52\\" y1=\\"32.52\\" x2=\\"76.85\\" y2=\\"76.85\\" data-v-57536c88=\\"\\"></line>
              <line class=\\"cls-2\\" x1=\\"76.85\\" y1=\\"32.52\\" x2=\\"32.52\\" y2=\\"76.85\\" data-v-57536c88=\\"\\"></line>
            </g>
          </g>
        </g>
      </svg></div>
  </div>
  <div class=\\"grid grid-cols-8 gap-4 border-b-2\\">
    <div></div>
    <div class=\\"col-span-5 ml-4\\">pTzf9MBAJLPbn4ZeLX80bB3</div>
    <div class=\\"justify-self-center\\">
      <div></div>
    </div>
    <div class=\\"h-full justify-self-center\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 109.38 109.38\\" class=\\"w-5 h-full\\" data-v-34d8da4e=\\"\\">
        <g data-v-34d8da4e=\\"\\">
          <g data-v-34d8da4e=\\"\\">
            <circle class=\\"cls-1\\" cx=\\"54.69\\" cy=\\"54.69\\" r=\\"54.69\\" data-v-34d8da4e=\\"\\"></circle>
            <polyline class=\\"cls-2\\" points=\\"22.32 52.84 49.06 76.32 87.06 33.05\\" data-v-34d8da4e=\\"\\"></polyline>
          </g>
        </g>
      </svg></div>
  </div>
  <div class=\\"grid grid-cols-8 gap-4 border-b-2\\">
    <div></div>
    <div class=\\"col-span-5 ml-4\\">gOt73aQs</div>
    <div class=\\"justify-self-center\\">
      <div></div>
    </div>
    <div class=\\"h-full justify-self-center\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 109.38 109.38\\" class=\\"w-5 h-full\\" data-v-34d8da4e=\\"\\">
        <g data-v-34d8da4e=\\"\\">
          <g data-v-34d8da4e=\\"\\">
            <circle class=\\"cls-1\\" cx=\\"54.69\\" cy=\\"54.69\\" r=\\"54.69\\" data-v-34d8da4e=\\"\\"></circle>
            <polyline class=\\"cls-2\\" points=\\"22.32 52.84 49.06 76.32 87.06 33.05\\" data-v-34d8da4e=\\"\\"></polyline>
          </g>
        </g>
      </svg></div>
  </div>
</div>"
`;
